docs/topics/compatibility-guide-17.md

```diff
@@ -0,0 +1,489 @@
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: deprecate the `kapt.use.worker.api` property that allowed to run kapt via Gradle Workers API (default: true)
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: raise the deprecation level to a warning
+> - \>= 1.8.0: remove this property
+
+### Remove kotlin.experimental.coroutines Gradle DSL option and kotlin.coroutines Gradle property
+
+> **Issue**: [KT-50494](https://youtrack.jetbrains.com/issue/KT-50494)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `kotlin.experimental.coroutines` Gradle DSL option and the `kotlin.coroutines` property
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: raise the deprecation level to a warning
+> - 1.7.0: remove the DSL option, its enclosing `experimental` block, and the property
+
+### Deprecate useExperimentalAnnotation compiler option
+
+> **Issue**: [KT-47763](https://youtrack.jetbrains.com/issue/KT-47763)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the hidden `useExperimentalAnnotation()` Gradle function used to opt in to using an API in a module.
+> `optIn()` function can be used instead
+> 
+> **Deprecation cycle:**
+> 
+> - 1.6.0: hide the deprecation option
+> - 1.7.0: remove the deprecated option
+
+### Deprecate kotlin.compiler.execution.strategy system property
+
+> **Issue**: [KT-51830](https://youtrack.jetbrains.com/issue/KT-51830)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: deprecate the `kotlin.compiler.execution.strategy` system property used to choose a compiler execution strategy. 
+> `use the Gradle property `kotlin.compiler.execution.strategy` or the compile task property `compilerExecutionStrategy` instead
+>
+> **Deprecation cycle:**
+>
+> - 1.7.0: raise the deprecation level to a warning
+> - \> 1.7.0: remove the property
+
+### Remove kotlinOptions.jdkHome compiler option
+
+> **Issue**: [KT-46541](https://youtrack.jetbrains.com/issue/KT-46541)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `kotlinOptions.jdkHome` compiler option used to include a custom JDK from the specified location 
+> into the classpath instead of the default `JAVA_HOME`. Use [Java toolchains](gradle.md#gradle-java-toolchains-support) instead
+>
+> **Deprecation cycle:**
+>
+> - 1.5.30: raise the deprecation level to a warning
+> - \> 1.7.0: remove the option
+
+### Remove noStdlib compiler option
+
+> **Issue**: [KT-49011](https://youtrack.jetbrains.com/issue/KT-49011)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `noStdlib` compiler option. The Gradle plugin uses the `kotlin.stdlib.default.dependency=true` property to control the Kotlin standard library presence
+>
+> **Deprecation cycle:**
+>
+> - 1.5.0: raise the deprecation level to a warning
```

Comment:
Note: usually deprecation level is _raised_ when it was a warning and becomes an error. When the warning appears first time, we say that we introduce a warning, report a warning, and so on.

Reply:
Ok, I'll fix in my Tools features

Reply:
Actually, in the previos compatibility guide we used the following wording:
> - 1.5.0: raise the deprecation level to a warning

> - 1.5.0: raise the deprecation level to an error

Maybe we should keep in mind your note for future compatibility guides.